### PR TITLE
Add sample game generators from bimatrix-generators

### LINKED ISSUE #53: bimatrix_generators
We have got a permission from the corresponding author of the paper "[An Empirical Study of Finding Approximate Equilibria in Bimatrix Games](https://arxiv.org/abs/1502.04980)" to include the classes of games studied there into our libraries `Games.jl` and `quantecon.game_theory`. The [associated C code](https://github.com/bimatrix-games/bimatrix-generators) is now distributed under BSD3 license.

* [x] Colonel Blotto Games
* [x] Ranking Games
* [x] SGC Games
* [x] Tournament Games
* [x] Unit vector Games

### PR DESCRIPTION
**Pull request by oyamad**
Close #53. Julia version of QuantEcon/QuantEcon.py#392.

### CODE REVIEW

File: src/generators/bimatrix_generators.jl [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+for player 1 is 1 if, in the tournament, the node chosen by player 1 dominates
+all the nodes in the k-subset chosen by player 2. The payoff for player 2 is 1
+if player 2's k-subset contains player 1's node.
+
+# Notes
+
+The actions of player 2 are ordered according to the
+[combinatorial number system]
+(https://en.wikipedia.org/wiki/Combinatorial_number_system),
+different from the order used in the original library in C.
```
Comment (QBatista) [inline src/generators/bimatrix_generators.jl]: 'which is different'

File: src/generators/bimatrix_generators.jl [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+    tournament_game(n, k; seed=-1)
+
+Return a NormalFormGame instance of the 2-player win-lose game, whose payoffs
+are either 0 or 1, introduced by Anbalagan et al. (2013). Player 1 has n
+actions, which constitute the set of nodes {1, ..., n}, while player 2 has n
+choose k actions, each corresponding to a subset of k elements of the set of n
+nodes. Given a randomly generated tournament graph on the n nodes, the payoff
+for player 1 is 1 if, in the tournament, the node chosen by player 1 dominates
+all the nodes in the k-subset chosen by player 2. The payoff for player 2 is 1
+if player 2's k-subset contains player 1's node.
```
Comment (QBatista) [inline src/generators/bimatrix_generators.jl]: What about 'player 1's chosen node'?

File: test/generators/test_bimatrix_generators.jl [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            g2 = blotto_game(MersenneTwister(32), h, t, rho)
+            for i in 1:2
+                @test g1.players[i].payoff_array == g2.players[i].payoff_array
+            end
+        end
+
+        @testset "test_payoff_generation" begin
+            # WARNING: These results were computed by hand and depend on the
+            # structure of the function being tested. Altering the structure of
+            # the function will cause this test to fail.
```
Comment (QBatista) [inline test/generators/test_bimatrix_generators.jl]: Is this warning still necessary after refactorizing?

File: src/generators/bimatrix_generators.jl [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+  the higher score wins the prize. Each player's payoff is given by the value of
+  the prize minus the cost of the effort.
+
+* SGC Games (`sgc_game`): These games were introduced by Sandholm, Gilpin, and
+  Conitzer (2005) as a worst case scenario for support enumeration as it has a
+  unique equilibrium where each player uses half of his actions in his support.
+
+* Tournament Games (`tournament_game`): These games are constructed by
+  Anbalagan et al. (2013) as games that do not have interim epsilon-Nash
+  equilibria with constant cardinaliry supports for epsilon smaller than a
```
[the remark this answers is not in the document]
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Corrected.

File: src/generators/bimatrix_generators.jl [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+  certain threshold.
+
+* Unit vector Games (`unit_vector_game`): These games are games where the payoff
```
[the remark this answers is not in the document]
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Corrected.

File: src/generators/bimatrix_generators.jl [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+  matrix of one player consists of unit (column) vectors, used by Savani and von
+  Stengel (2016) to construct instances that are hard, in temrs of computation
```
[the remark this answers is not in the document]
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Corrected.

File: src/generators/bimatrix_generators.jl [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+  complexity, both for the Lemke-Howson and support enumeration algorithms.
+
+Large part of the code here is based on the C code available at
```
[the remark this answers is not in the document]
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Uncountable sounds right to me. We should ask a native English speaker.

What do you think @mmcky?

File: src/generators/bimatrix_generators.jl [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+"""
+    ranking_game([rng=GLOBAL_RNG], n[, steps=10])
+
+Return a NormalFormGame instance of (the 2-player version of) the "ranking game"
+studied by Goldberg et al. (2013), where each player chooses an effort level
+associated with a score and a cost which are both increasing functions with
+randomly generated step sizes. The player with the higher score wins the first
+prize, whose value is 1, and the other player obtains the "second prize" of
+value 0; in the case of a tie, the first prize is split and each player receives
+a value 0.5. The payoff of a player is given by the value of the prize minus the
```
[the remark this answers is not in the document]
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Corrected.

File: src/generators/bimatrix_generators.jl (hunk 1 above)
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Just "different" sounds fine to me.

File: src/generators/bimatrix_generators.jl [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+cost of the effort.
+
+# Arguments
+
+- `rng::AbstractRNG=GLOBAL_RNG`: Random number generator used.
+- `n::Integer` : Number of actions, i.e, number of possible effort levels.
+- `steps::Integer=10` : Parameter determining the random step sizes for the
+  scores and costs for each player: The step sizes for the scores are drawn from
```
[the remark this answers is not in the document]
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: That's much better.

File: src/generators/bimatrix_generators.jl (hunk 2 above)
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Incorporated.

File: test/generators/test_bimatrix_generators.jl (hunk 3 above)
Comment (oyamad) [reply test/generators/test_bimatrix_generators.jl]: Yes, it think so. There may be further refactoring in the future.

File: src/generators/bimatrix_generators.jl (hunk 1 above)
Comment (mmcky) [reply src/generators/bimatrix_generators.jl]: I would vote for `which is different`

File: test/generators/test_bimatrix_generators.jl (hunk 3 above)
Comment (mmcky) [reply test/generators/test_bimatrix_generators.jl]: I think this is a good warning to have.

File: src/generators/bimatrix_generators.jl (hunk 1 above)
Comment (oyamad) [reply src/generators/bimatrix_generators.jl]: Thanks. Incorporated.

**Pull request merged**